kiwi/boot/image/dracut.py

```diff
@@ -226,14 +243,14 @@ def get_boot_names(self):
                 self.boot_root_directory
             )
         dracut_output_format = self._get_dracut_output_file_format()
-        return boot_names_type(
+        return BootNames(
             kernel_name=kernel_info.name,
             initrd_name=dracut_output_format.format(
                 kernel_version=kernel_info.version
             )
         )
 
-    def _get_dracut_output_file_format(self):
+    def _get_dracut_output_file_format(self) -> str:
```

Comment:
All type hinting we add I think should be done only for the public interface. That's the part where I think it would make sense. People using kiwi as an api and running our methods benefit from type checking. Private methods are not considered to be called and we are free to change them whenever we want. Type hinting there imho just makes it more complex

Reply:
> Private methods are not considered to be called and we are free to change them whenever we want.

Exactly. And if we change the signature, then we should get a warning/error when a caller of that function violates the function's contract (e.g. the return type), which type hints will provide us with. That's why I've added them to private functions too, not because I like to grind :wink:

Reply:
ok got it, looking from a developer perspective make sense